Competition2017/src/competition/subsystems/agitator/AgitatorSubsystem.java

```diff
@@ -1,43 +1,135 @@
 package competition.subsystems.agitator;
 
-import javax.inject.Singleton;
-
 import org.apache.log4j.Logger;
-
+import com.ctre.CANTalon.FeedbackDevice;
 import com.ctre.CANTalon.TalonControlMode;
-import com.google.inject.Inject;
-
+import competition.subsystems.RobotSide;
+import edu.wpi.first.wpilibj.Timer;
 import xbot.common.command.BaseSubsystem;
 import xbot.common.controls.actuators.XCANTalon;
 import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.PIDPropertyManager;
+import xbot.common.properties.BooleanProperty;
 import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.XPropertyManager;
-@Singleton
-public class AgitatorSubsystem extends BaseSubsystem{
+
+public class AgitatorSubsystem extends BaseSubsystem {
     private static Logger log = Logger.getLogger(AgitatorSubsystem.class);
-    
-    public final DoubleProperty motorPowerProperty;
-    public final XCANTalon agitatorMotor;
-    
-    @Inject
-    public AgitatorSubsystem(WPIFactory factory, XPropertyManager propManager){
-        log.info("Creating Agitator Subsystem");
-        agitatorMotor = factory.getCANTalonSpeedController(6);
-        agitatorMotor.setBrakeEnableDuringNeutral(true);
+    protected final XCANTalon agitatorMotor;
+    private final RobotSide side;
+    protected final DoubleProperty intakePowerProperty;
+    protected final DoubleProperty ejectPowerProperty;
+    protected final DoubleProperty agitatorTargetSpeed;
+    private final DoubleProperty agitatorCurrentSpeed;
+    private final DoubleProperty agitatorOutputPower;
+    private final DoubleProperty agitatorTalonError;
+    private final BooleanProperty atSpeedProp;
+    private final BooleanProperty enableAgitatorLogging;
+    private final DoubleProperty agitatorSpeedThresh;
+    private final PIDPropertyManager pidProperties;
+
+    public AgitatorSubsystem(int motor, RobotSide side, WPIFactory factory, XPropertyManager propManager, RobotAssertionManager assertionManager){
+        log.info("Creating " + side + " Agitator");
+
+        this.side = side;
+        this.pidProperties = new PIDPropertyManager("AgitatorSpeed", propManager, assertionManager, 0, 0, 0, 0);
+
+        agitatorSpeedThresh = propManager.createPersistentProperty(side + "Agitator nominal speed thresh (TPC)", motor);
+        agitatorCurrentSpeed = propManager.createEphemeralProperty(side + "Agitator current speed (TPC)", 0);
+        agitatorTargetSpeed = propManager.createEphemeralProperty(side + "Agitator goal speed (TPC)", 0);
+        agitatorOutputPower = propManager.createEphemeralProperty(side + "Agitator voltage", 0);
+        intakePowerProperty = propManager.createPersistentProperty("Agitator intake power", 0.5);
```

Comment:
Talked with Alex & Seth, we agreed using a simple power-controlled system makes more sense for now.